packages/core/src/dialects/mssql/query.js

```diff
@@ -323,65 +319,54 @@ export class MsSqlQuery extends AbstractQuery {
         ));
       });
 
-      return new sequelizeErrors.UniqueConstraintError({ message, errors, cause: err, fields });
+      const uniqueConstraintError = new sequelizeErrors.UniqueConstraintError({ message, errors, cause: err, fields });
+      if (err.errors?.length > 0) {
+        return new sequelizeErrors.AggregateError([...err.errors, uniqueConstraintError]);
+      }
+
+      return uniqueConstraintError;
     }
 
-    match = err.message.match(/Failed on step '(.*)'.Could not create constraint. See previous errors./)
```

Comment:
I removed this as I cant seem to find what triggers this and it doesn't have the same level of information for a `ForeignKeyConstraintError`.
This should be returned as an `UnknownConstraintError` inside a `AggregateError`